packages/zwave-js/src/lib/commandclass/BarrierOperatorCC.ts

```diff
@@ -141,6 +147,72 @@ export class BarrierOperatorCCAPI extends CCAPI {
 		}
 	}
 
+	public async getCapabilities(): Promise<readonly SignalType[]> {
+		this.assertSupportsCommand(
+			BarrierOperatorCommand,
+			BarrierOperatorCommand.CapabilitiesGet,
+		);
+
+		const cc = new BarrierOperatorCCCapabilitiesGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+		});
+		const response = (await this.driver.sendCommand<BarrierOperatorCCCapabilitiesReport>(
+			cc,
+			this.commandOptions,
+		))!;
+		return response.supportedSignalTypes;
+	}
+
+	// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+	public async getEvent(signalType: SignalType) {
+		this.assertSupportsCommand(
+			BarrierOperatorCommand,
+			BarrierOperatorCommand.EventGet,
+		);
+
+		const cc = new BarrierOperatorCCEventGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			signalType,
+		});
+		const response = (await this.driver.sendCommand<BarrierOperatorCCEventReport>(
+			cc,
+			this.commandOptions,
+		))!;
+		return {
+			signalType: response.type,
+			signalState: response.state,
+		};
```

Comment:
I think you're retrieving the signalState for a signalType. Thus the method should only return the signalState.

Reply:
you right! enumerated it also